test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,126 @@
+    return (resultList)
+
+def get_region_list(dataFile):
+    ''' Retrieve topNRegions/functions with the highest cyclomatic complexity values. '''
+    top5 = dataFrame.nlargest(topNRegions, 'std.code.complexity:cyclomatic')
+    atlasFormat = {}
+    for index, row in top5.iterrows():
+        atlasFormat[row["region"]] = row["std.code.complexity:cyclomatic"]
+
+    return (atlasFormat)
+
+def get_complexity_ranges_list(dataFile):
+    ''' Retrieve 3 set of cyclomatic complexity ranges, above 30, above 50 and above 90. '''
+    rangesList = [20, 50, 90]
+    atlasFormat = {}
+    for i in rangesList:
+        aboveRangeString = 'Above ' + str(i)
+        atlasFormat[aboveRangeString] = get_complexity_max_limit(dataFile, i)
+
+    return (atlasFormat)
+
+def get_complexity_max_limit(complexity_data_path: str, max_limit:int):
+    ''' Retrieve the number of cyclomatic complexity values above the max_limit. '''
+    columnName = 'std.code.complexity:cyclomatic'
+    # Select column 'std.code.complexity:cyclomatic' from the dataframe
+    column = dataFrame[columnName]
+    # Get count of values greater than max_limit in the column 'std.code.complexity:cyclomatic'
+    count = column[column > max_limit].count()
+    """
+    Pandas aggregation functions (like sum, count and mean) returns a NumPy int64 type number not a Python integer.
+    Object of type int64 is not JSON serializable so need to convert into an int.
+    """
+    return (int(count))
+
+def get_average(complexity_summary_file):
+    ''' Retrieve the cyclomatic code complexity value '''
+    with open(complexity_summary_file, 'r') as f:
+        data = f.read()
+
+    # Below is the format of complexity_summary_file, we are intersted in extracting the "avg" value of std.code.complexity.
+    """
+    {
+        "view": [
+            {
+                "data": {
+                    "info": {"path": "./", "id": 1},
+                    "aggregated-data": {
+                        "std.code.complexity": {
+                            "cyclomatic": {
+                                "max": 91,
+                                "min": 0,
+                                "avg": 4.942372881355932,
+                                "total": 14580.0,
+                                "count": 2950,
+                                "nonzero": False,
+                                "distribution-bars": [ ],
+                                "sup": 0,
+                            }
+                        },
+                        "std.code.lines": {
+                            "code": { }
+                        },
+                    },
+                    "file-data": {},
+                    "subdirs": [ ],
+                    "subfiles": [ ],
+                }
+            }
+        ]
+    }
+    """
+
+    d = ast.literal_eval(data)
+    newDictionary = d["view"]
+
+    return (newDictionary[0]['data']['aggregated-data']['std.code.complexity']['cyclomatic']['avg'])
```

Comment:
It's safer to include this section of code in a try exception block, to capture miss-formated data files. 
Also, suggest returning a variable instead of the super-long statement.